lib/teiserver_web/controllers/moderation/report_form_controller.ex

```diff
@@ -47,63 +47,63 @@ defmodule TeiserverWeb.Moderation.ReportFormController do
   @spec create(Plug.Conn.t(), Map.t()) :: Plug.Conn.t()
   def create(conn, %{"report" => report}) do
     target_id = report["target_id"] |> int_parse
-
-    {match_id, relationship} =
-      case report["match_id"] do
-        "none" ->
-          {nil, nil}
-
-        match_id_str ->
-          case Battle.get_match(match_id_str, preload: [:members]) do
-            nil ->
-              {nil, nil}
-
-            match ->
-              target_member =
-                match.members
-                |> Enum.find(fn member -> member.user_id == target_id end)
-
-              reporter_member =
-                match.members
-                |> Enum.find(fn member -> member.user_id == conn.assigns.current_user.id end)
-
-              relationship =
-                cond do
-                  reporter_member == nil -> nil
-                  target_member.team_id == reporter_member.team_id -> "Allies"
-                  target_member.team_id != reporter_member.team_id -> "Opponents"
-                end
-
-              {match.id, relationship}
-          end
+    if conn.assigns.current_user.id == target_id do
+      conn
+      |> put_status(:unprocessable_entity)
+      |> render("self_report.html", message: "Reporter and target may not be the same user.")
```

Comment:
While the statement is accurate, it doesn't read well. I'd rather something like "You cannot report yourself", we are speaking to the reporter themselves rather than a 3rd party trying to enter it into an admin control panel.